Tasks/JavaToolInstallerV0/FileExtractor/JavaFilesExtractor.ts

```diff
@@ -27,31 +28,32 @@ export class JavaFilesExtractor {
         }
     }
 
-    private async unzipExtract(file, destinationFolder) {
-        console.log(taskLib.loc('UnzipExtractFile', file));
-        if (process.platform === 'win32') {
-            // build the powershell command
-            const escapedFile = file.replace(/'/g, "''").replace(/"|\n|\r/g, ''); // double-up single quotes, remove double quotes and newlines
-            const escapedDest = destinationFolder.replace(/'/g, "''").replace(/"|\n|\r/g, '');
-            const command: string = `$ErrorActionPreference = 'Stop' ; try { Add-Type -AssemblyName System.IO.Compression.FileSystem } catch { } ; [System.IO.Compression.ZipFile]::ExtractToDirectory('${escapedFile}', '${escapedDest}')`;
-    
-            // change the console output code page to UTF-8.
-            const chcpPath = path.join(process.env.windir, "system32", "chcp.com");
-            await taskLib.exec(chcpPath, '65001');
-    
-            // run powershell
-            const powershell: tr.ToolRunner = taskLib.tool('powershell')
-                .line('-NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command')
-                .arg(command);
-            await powershell.exec();
-        } else {
-            const unzip: tr.ToolRunner = taskLib.tool('unzip')
-                .arg(file);
-            await unzip.exec(<tr.IExecOptions>{ cwd: destinationFolder });
-        }
+    private isTar(file) {
+        var name = this.win ? file.toLowerCase() : file;
```

Comment:
Use `const`.

Also above, can you change `win` to be a readonly boolean instead of RegExpMatchArray and initialize it like this?
```ts
this.win = os.platform() === 'win32';
```